Microsoft.Toolkit.Uwp.UI/Extensions/Mouse/Mouse.cs

```diff
@@ -0,0 +1,114 @@
+{
+   /// <summary>
+   /// Helper class for easily changing the mouseover cursor type.
+   /// </summary>
+   public class Mouse : DependencyObject
+   {
+      /// <summary>
+      /// Dependency property for specifying the target <see cref="CoreCursorType"/> to be shown
+      /// over the target <see cref="FrameworkElement"/>.
+      /// </summary>
+      public static readonly DependencyProperty CursorProperty =
+          DependencyProperty.RegisterAttached("Cursor", typeof(CoreCursorType), typeof(Mouse), new PropertyMetadata(CoreCursorType.Arrow, CursorChanged));
+
+      /// <summary>
+      /// Set the target <see cref="CoreCursorType"/>.
+      /// </summary>
+      /// <param name="element">Object where the selector cursor type should be shown.</param>
+      /// <param name="value">Target cursor type value.</param>
+      public static void SetCursor(FrameworkElement element, CoreCursorType value)
+      {
+         element.SetValue(CursorProperty, value);
+      }
+
+      /// <summary>
+      /// Get the current <see cref="CoreCursorType"/>.
+      /// </summary>
+      /// <param name="element">Object where the selector cursor type should be shown.</param>
+      /// <returns>Cursor type set on target element.</returns>
+      public static CoreCursorType GetCursor(FrameworkElement element)
+      {
+         return (CoreCursorType)element.GetValue(CursorProperty);
+      }
+
+      private static void CursorChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+      {
+         FrameworkElement element = d as FrameworkElement;
+         if (element == null)
+         {
+            throw new NullReferenceException(nameof(element));
+         }
+
+         CoreCursorType value = (CoreCursorType)e.NewValue;
+
+         lock (CursorLock)
+         {
+            if (!Cursors.ContainsKey(value))
+            {
+               Cursors[value] = new CoreCursor(value, 1);
+            }
+
+            // make sure event handlers are not attached twice to element
+            element.PointerEntered -= Element_PointerEntered;
+            element.PointerEntered += Element_PointerEntered;
+            element.PointerExited -= Element_PointerExited;
+            element.PointerExited += Element_PointerExited;
+            element.Unloaded -= ElementOnUnloaded;
+            element.Unloaded += ElementOnUnloaded;
+         }
+      }
+
+      private static void Element_PointerEntered(object sender, PointerRoutedEventArgs e)
+      {
+         CoreCursorType cursor = GetCursor((FrameworkElement)sender);
+         Window.Current.CoreWindow.PointerCursor = Cursors[cursor];
+      }
+
+      private static void Element_PointerExited(object sender, PointerRoutedEventArgs e)
+      {
+         // when exiting change the cursor to the target Mouse.Cursor value of the new element
+         CoreCursor cursor;
+         if (e.OriginalSource is FrameworkElement newElement)
+         {
+            cursor = Cursors[GetCursor(newElement)];
```

Comment:
What if this new element doesn't have a cursor defined?

Reply:
Default value for this attached DependencyProperty is CoreCursorType.Arrow.